mu4/instruments/view/instrumentlistmodel.cpp

```diff
@@ -204,7 +204,7 @@ void InstrumentListModel::unselectInstrument(const QString& id)
 
 void InstrumentListModel::swapSelectedInstruments(int firstIndex, int secondIndex)
 {
-    m_selectedInstruments.swap(firstIndex, secondIndex);
+    m_selectedInstruments.swapItemsAt(firstIndex, secondIndex);
```

Comment:
Requires Qt 5.13 or later, so better protect this with a corresponding `#if'

Reply:
MU4 will use Qt 5.15. This is the final decision :)

Reply:
@Jojo-Schmitz Could you tell me if you think it makes sense to support builds for Qt 5.12, 5.13?

Reply:
5.13 may not make much sense, 5.12 might

Reply:
And what could be the reasons for maintaining the Qr 5.12 build?

Reply:
> MU4 will use Qt 5.15. This is the final decision :)

Would have been nice it that had been discissed/published in the developers' chat

Reply:
> macOS 10.10 and 10.11
Yes, it was a limiting factor for transferring MuseScore 3 to Qt higher than 5.9

But Apple itself does not support these platforms.
We leave for them MuseScore 3

And we think that they should not hinder the development of MuseScore 4

Reply:
It doesn't hinder so far, if done properly. So far we don't really need any features newer than Qt 5.9 as far as I can tell?

Reply:
Qt 5.15 supports macOS 10.13 (https://doc.qt.io/qt-5/supported-platforms.html), it is still supported by Apple.
MacOS 10.12 is not supported by Apple since October 2019 (https://en.wikipedia.org/wiki/MacOS_Sierra)

Reply:
So we're loosing 3 macOS versions on the way. Mac users won't be amused...

Reply:
> It doesn't hinder so far, if done properly. So far we don't really need any features newer than Qt 5.9 as far as I can tell?
> 

Our main motivation for moving to Qt 5.15 now is that there are a lot of changes in Qt 5.15 compared to Qt 5.9, especially with relationship to Qml.
And this is preparatory, the latest version for Qt 6.0

Now there is an active development (and it will still last at least six months). We can detect and fix all the problems associated with the transition to Qt 5.15 (and there can be many because there are many changes).
If we decide to switch to Qt 5.15 after the release of MuseScore 4, then our users will face all these problems.

Reply:
Yes, sure, using Qt 5.15 for MuseScore 4 is fine, but if and as long as we can keep compatibility with Qt 5.9 with a reasonably effort, that'd be even better. And so far I don't see any reason not to.

Reply:
Unfortunately, the effort is more than desirable.
There will be a lot of `#ifdef` in the code
But even worse, as for Qml, there are many syntax changes.
And support for Qt 5.9 means that we cannot use any of the new features Qt 5.15

Reply:
Quite a few of those `#ifdef`s are in the code already.
What new featured are we using then?

Reply:
![image](https://user-images.githubusercontent.com/3818029/94917714-5ebb7c80-04b1-11eb-8865-d3d7291be790.png)
https://gs.statcounter.com/macos-version-market-share/desktop/worldwide
These are not MuseScore statistics, but statistics on the use of the OS in general

MacOS 10.10 - 2.43%
MacOS 10.11 - 3.82%
MacOS 10.12 - 4.8%

When MuseScore 4 is released (not earlier than half a year), the use of these versions will be even less.

Reply:
Loosing ~11% of the Mac uses is quite a number

Reply:
But we don't lose them :)
They will have MuseScore 3.6

Reply:
Tell them and take the heat ;-)